### DISCUSSION TITLE
Theia build failed at yarn command with exit code 1

### DISCUSSION THREAD

**Discussion by bahuf** [eclipse-theia/theia#10616] [Q&A]
Hey everyone,
I am trying to build Theia on Windows.
I did all these steps:
1. Install scoop
2. Installed nvm with scoop
3. Installed Node.js version 12.14.1 with nvm
4. Installed yarn with scoop
5. Installed windows-building-tools from visual studio
6. Installed Python 3.10.1 via website
7. Then I created a folder on my desktop and opened git bash
8. Then I tried to use this command: 
git clone https://github.com/eclipse-theia/theia.git \
    && cd theia \
    && yarn \
    && yarn browser build \
    && yarn browser start
The git clone works fine but the build fails at yarn with the error shown below.
Does anyone know a fix for this.

![image](https://user-images.githubusercontent.com/92301607/148925227-c8e670f2-fbdb-4446-9eab-1fbf3838ed68.png)
![image](https://user-images.githubusercontent.com/92301607/148925363-3b2d6e7f-46fd-4984-856e-fbfd563f37f5.png)

**Comment by msujew** [comment]
Hi @bahuf,

it seems like the windows build tools didn't install themselves correctly (from your image: `Error: MSBuild is not set, please run 'node-gyp configure'`

You did say you used the Visual Studio installer? Can you try the `npm` command for installing these tools mentioned in the [development guide](https://github.com/eclipse-theia/theia/blob/master/doc/Developing.md#building-on-windows)?

**Reply by bahuf** [reply]
Hi @msujew,
thanks for your quick reply.
I uninstalled Windows Build Tools 2017 and Visual Studio 2017 through the Visual Studio installer.
Then I used the command "npm --add-python-to-path install --global --production windows-build-tools", that was mentioned in the development guide, but the Windows Tools Installer got stuck (Python got installed successfully)
![vs2017](https://user-images.githubusercontent.com/92301607/148939878-ca65ff06-ecfe-4adc-bb74-fa06c8b6c568.png)

After some research I used the command for Windows Build Tools 2015 "npm install --global --production windows-build-tools --vs2015". The Installer ran through for both installations this time.
![2022-01-11 12_57_46-](https://user-images.githubusercontent.com/92301607/148940269-94956b16-b31b-4f7f-9c62-2a79ecfe1f88.png)

But now yarn doesn't seem to recognize the Windows Build Tools.
![yarn 2015](https://user-images.githubusercontent.com/92301607/148940421-dcbe6bdd-0d66-4047-adf7-de9cac3aadf9.png)

**Comment by bahuf** [comment]
I couldn't find an anwser to my problem, therefore I installed a VM with windows on my pc and reproduced all the steps in the development guide. In the VM everything worked fine. I know this isn't the best solution, but as long as i don't know any other fix for this problem, this is working for me.
